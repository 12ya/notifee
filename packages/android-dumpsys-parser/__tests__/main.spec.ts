/* tslint:disable:non-literal-fs-path */
import { default as parser, stringToHashCode } from './../build/index.min';
import * as fs from 'fs';
import * as path from 'path';

describe('exports', () => {
  it('export the parser as default', () => {
    expect(parser).toBeInstanceOf(Function);
  });

  it('exports the stringToHashCode utility', () => {
    expect(stringToHashCode).toBeInstanceOf(Function);
  });
});

let sample;

describe('Android 28+', () => {
  beforeAll(async () => {
    sample = fs
      .readFileSync(path.resolve(__dirname, './samples/android_28.txt'))
      .toString('utf-8');
  });

  it('returns parsed notification records in date descending order (newest first)', () => {
    const output = parser(sample);
    expect(output).toBeTruthy();
    expect(output).toBeInstanceOf(Array);
    expect(output.length).toBe(4);

    let timestampPrevious = 999999999999999;
    for (let i = 0; i < output.length; i++) {
      const outputElement = output[i];
      expect(outputElement.creationTimeMs).toBeLessThan(timestampPrevious);
      timestampPrevious = outputElement.creationTimeMs;
    }
  });

  describe('extras', () => {
    it('reads android. prefixed extras and places them on root object', () => {
      const output = parser(sample);
      expect(output).toBeTruthy();
      expect(output[0]).toMatchSnapshot();
    });
  });
});
