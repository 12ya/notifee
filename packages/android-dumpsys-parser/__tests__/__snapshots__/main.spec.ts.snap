// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Android 28+ extras reads android. prefixed extras and places them on root object 1`] = `
Object {
  "actions": Array [],
  "appInfo": "ApplicationInfo{7f0f055 com.invertase.testing}",
  "asstImportance": "LOW",
  "attributes": "AudioAttributes: usage=USAGE_NOTIFICATION content=CONTENT_TYPE_SONIFICATION flags=0x0 tags= bundle=null",
  "bigContentView": null,
  "channel": Object {
    "allowBubbles": true,
    "audioAttributes": null,
    "blockableSystem": false,
    "bypassDnd": false,
    "deleted": false,
    "description": null,
    "foregroundServiceShown": false,
    "group": null,
    "id": "foo",
    "importance": 3,
    "importanceLockedByOEM": false,
    "importanceLockedDefaultApp": false,
    "lightColor": 0,
    "lights": false,
    "lockscreenVisibility": -1000,
    "name": "Hello Foo",
    "showBadge": true,
    "sound": "content://settings/system/notification_sound",
    "userLockedFields": 0,
    "vibration": null,
    "vibrationEnabled": false,
  },
  "color": "0x00000000      timeout=unknown",
  "colorized": false,
  "contactAffinity": 0,
  "contentIntent": null,
  "contentView": null,
  "creationTimeMs": 1566810835037,
  "data": Object {
    "customValue1": "invertase",
    "customValue2": "salakar",
  },
  "deleteIntent": null,
  "flags": "0x10",
  "fullscreenIntent": null,
  "group": "foo bar group",
  "groupAlertBehavior": 0,
  "hasSeen": true,
  "headsUpContentView": null,
  "hidden": false,
  "icon": "com.invertase.testing:drawable/redbox_top_border_background",
  "id": -95542801,
  "importance": "LOW",
  "importanceExplanation": "asst",
  "infoText": null,
  "intercept": false,
  "interruptionTimeMs": 1566810835145,
  "isAppImportanceLocked": false,
  "isInterruptive": true,
  "largeIcon": null,
  "light": null,
  "number": 0,
  "packageName": "com.invertase.testing",
  "packagePriority": 0,
  "packageVisibility": -1000,
  "picture": null,
  "priority": 0,
  "progress": 0,
  "progressIndeterminate": false,
  "progressMax": 0,
  "rankingTimeMs": 1566810835005,
  "recentlyIntrusive": false,
  "reduced.images": true,
  "remoteInputHistory": Array [
    "Hello",
    "World",
  ],
  "showBadge": true,
  "showChronometer": false,
  "showWhen": false,
  "sound": "content://settings/system/notification_sound",
  "style": "BigPictureStyle",
  "subText": "World",
  "suppressedVisualEffects": 0,
  "systemImportance": "UNSPECIFIED",
  "tag": null,
  "text": "foo bar baz",
  "tickerText": null,
  "title": "",
  "updateTimeMs": 1566810835037,
  "vibration": null,
  "visibleSinceMs": 1566810835870,
}
`;
